dnscache.go

```diff
@@ -132,10 +149,13 @@ func (r *Resolver) lookupFunc(key string) func() (interface{}, error) {
 	if len(key) == 0 {
 		panic("lookupFunc with empty key")
 	}
-	resolver := net.DefaultResolver
+
+	// Deprecated check of the resolver attribute. Use the constructor to execute the check only once.
+	var resolver DNSResolver = net.DefaultResolver
```

Comment:
Let's not deprecate this. See my other comment about not providing a constructor.